src/libecalc/core/consumers/compressor/component.py

```diff
@@ -151,25 +145,19 @@ def evaluate(
             ),
             outlet_pressure_before_choking=outlet_pressure_before_choke,
             stages=[
-                Stage(
-                    name="inlet",
-                    stream=Stream(
-                        rate=total_requested_rate,
-                        pressure=operational_settings.inlet_pressure,
-                    ),
-                ),
+                Stage(name="inlet", stream=total_requested_inlet_stream),
                 Stage(
                     name="before_choke",
                     stream=Stream(
-                        rate=total_requested_rate,
+                        rate=total_requested_inlet_stream.rate,
                         pressure=outlet_pressure_before_choke,
                     ),
                 ),
                 Stage(
                     name="outlet",
                     stream=Stream(
-                        rate=total_requested_rate,
-                        pressure=operational_settings.outlet_pressure,
+                        rate=total_requested_inlet_stream.rate,  # Actual, not requested, different because of crossover
```

Comment:
supposed to be a TODO comment?

Reply:
and...inlet stream?

Reply:
No todo, the total_inlet_stream is the actual stream being sent through the compressor, while the requested might be a lot less since we have crossover.